### ISSUE TITLE
Solcast values zero and unknown - still have lots of API calls

### ISSUE THREAD

**Issue by BettySwallocks** [oziee/ha-solcast-solar#69]
Periodically my Solcast values seem to stop, show zeros and unknowns.  Unlike a previous issue I found I can confirm the daily API allowance is still well over zero when this happens.  I am in a development/trial HA solution so I am making changes quite a bit, including adding removing integrations and restarting the HA occasionally, so I can't rule that out as a potential link to this issue, however I can see it happening at times where my HA and Internet appears to be stable.

So I don't have a fully scientific and reproducible basis for the Solcast zero/unknown values, but I thought I would raise the issue and then see if anyone else is seeing similar.

**Comment by BettySwallocks** [comment]
Forgot to provide system details - development Pi3 hardware, limited memory

Home Assistant 2022.8.6
Supervisor 2022.08.3
Frontend 20220802.0 - latest

**Comment by craigrouse** [comment]
Same issue for me today. Not had a chance to look at the cause yet, but it stopped working suddenly.

**Comment by JoeMoer222** [comment]
I am running HA in a Debian 11 VM on a Truenas intallation and find that if the Truenas machine reboots for any reason that Solcast presents the same fault. See the attached image
![Solcast_error](https://user-images.githubusercontent.com/69679339/185868818-2dc3f89d-58cb-4073-b21d-defe59c06f82.JPG)
The only way to get it working again is to regenerate a new API key on the Solcast web site and re-configure the integration.

Home Assistant 2022.8.6
Supervisor 2022.08.3
Frontend 20220802.0 - latest

**Comment by oziee** [comment]
ok.. I am not seeing anything like this on my side.

let me know if you find any other info. I know that for some when the SD card in things like a pi sbc can corrupt the solcast.json file and cause problems. deleting this file and restarting ha will create a new one

@JoeMoer222 shouldn't need to generate new API keys. the integration creates a HA/config/solcast.json file that stores some data and the last api poll date/time, so it will only run like every hour.

**Comment by killee** [comment]
Have the same issue.
Deleting HA/config/solcast.json and restart HA doesn't help.
I never get the data from solcast 😔.

Attached some information.

```

  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2022.8.7",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.5",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/Berlin",
    "os_name": "Linux",
    "os_version": "5.15.60",
    "supervisor": "2022.08.3",
    "host_os": "Home Assistant OS 8.5",
    "docker_version": "20.10.14",
    "chassis": "vm",
    "run_as_root": true
  },
  "custom_components": {
    "ocpp": {
      "version": "v0.4.33",
      "requirements": [
        "ocpp>=0.14.1",
        "websockets>=10.2"
      ]
    },
    "easycontrols": {
      "version": "0.5.7",
      "requirements": [
        "eazyctrl==0.4"
      ]
    },
    "localtuya": {
      "version": "4.0.1",
      "requirements": []
    },
    "huawei_solar": {
      "version": "1.1.2",
      "requirements": [
        "pyserial==3.5",
        "pyserial-asyncio==0.6",
        "huawei-solar==2.1.3"
      ]
    },
    "solcast_solar": {
      "version": "3.0.19",
      "requirements": [
        "aiohttp>=3.6.2",
        "datetime>=4.3",
        "isodate>=0.6.0"
      ]
    },
    "scheduler": {
      "version": "v0.0.0",
      "requirements": []
    },
    "blnet": {
      "version": "0.5.0",
      "requirements": [
        "pyblnet==0.9.1"
      ]
    },
    "deepstack_face": {
      "version": "0.9.0",
      "requirements": [
        "pillow",
        "deepstack-python==0.8"
      ]
    },
    "blitzortung": {
      "version": "1.0.1",
      "requirements": [
        "paho-mqtt>=1.5.0"
      ]
    },
    "hacs": {
      "version": "1.26.2",
      "requirements": [
        "aiogithubapi>=22.2.4"
      ]
    },
    "alexa_media": {
      "version": "4.1.0",
      "requirements": [
        "alexapy==1.26.1",
        "packaging>=20.3",
        "wrapt>=1.12.1"
      ]
    },
    "frigate": {
      "version": "2.3",
      "requirements": []
    },
    "dwd_weather": {
      "version": "1.2.22",
      "requirements": [
        "simple_dwd_weatherforecast==1.1.5",
        "markdownify==0.6.5"
      ]
    },
    "candy": {
      "version": "1.0.0",
      "requirements": [
        "backoff>=1.8.0,<2.0.0"
      ]
    }
  },
  "integration_manifest": {
    "domain": "solcast_solar",
    "name": "Solcast PV Forecast",
    "version": "3.0.19",
    "config_flow": true,
    "documentation": "https://github.com/oziee/ha-solcast-solar",
    "issue_tracker": "https://github.com/oziee/ha-solcast-solar/issues",
    "requirements": [
      "aiohttp>=3.6.2",
      "datetime>=4.3",
      "isodate>=0.6.0"
    ],
    "dependencies": [],
    "after_dependencies": [
      "http"
    ],
    "iot_class": "cloud_polling",
    "codeowners": [
      "@oziee"
    ],
    "is_built_in": false
  },
  "data": {
    "info": {
      "api_key": "**REDACTED**"
    },
    "data": {
      "forecasts": [],
      "energy": {},
      "api_used": 0,
      "last_updated": "2000-01-01T03:03:44.163097+00:00"
    }
  }
}
```

**Comment by oziee** [comment]
What is the system time on the computer? Is it right?

**Comment by killee** [comment]
> What is the system time on the computer? Is it right?

The time is correct, timezone Berlin

**Comment by oziee** [comment]
what if any messages are in the logs for the integration.. if there are none make sure that you have logging set to include the integration

**Comment by killee** [comment]
> what if any messages are in the logs for the integration.. if there are none make sure that you have logging set to include the integration

I'll do that after my vacation in a few days. At the cell phone it's a bit ....

Thanks for supporting 😃

**Comment by killee** [comment]
Hi @oziee 

after manual trigger solcast_update_all_forecasts I get the log:

`2022-09-01 15:19:02.056 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/solcast_solar/coordinator.py", line 145, in service_event_update
    for update_callback in self._listeners:
RuntimeError: dictionary changed size during iteration`

**Comment by oziee** [comment]
I disabled the service so there is something. It right.

Delete the whole ha/config/custom-comp../solcast directory and reinstall the integration

**Comment by killee** [comment]
> I disabled the service so there is something. It right.
> 
> Delete the whole ha/config/custom-comp../solcast directory and reinstall the integration

Tried it, same issue.

**Comment by oziee** [comment]
this is the same error as in #56 and this was fixed
you has have some cached code using old py files.

delete the `HA/config/custom_components/solcast_solar` directory
right after deleting this reboot HA

clear cache

once rebooted make sure that the `HA/config/custom_components/solcast_solar` is not there and reinstall the integration

I am running
```
Home Assistant 2022.8.7
Supervisor 2022.08.5
```
and no problems showing up this end

unless I get more error log info (or it happens to me) its a bit hard to debug :)

**Comment by killee** [comment]
Hi @oziee

didn't help.
I've tried to get more logging data for you:

`2022-09-06 19:06:59.315 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration solcast_solar which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-09-06 19:07:07.915 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 2: {'persistent_notification', 'easycontrols', 'solcast_solar', 'template', 'input_select', 'analytics', 'scene', 'luftdaten', 'hacs', 'tag', 'history', 'alexa_media', 'shelly', 'trace', 'esphome', 'camera', 'auth', 'number', 'energy', 'forecast_solar', 'scheduler', 'mobile_app', 'maxcube', 'co2signal', 'ffmpeg', 'my', 'device_automation', 'blueprint', 'system_health', 'person', 'media_source', 'onboarding', 'dlna_dmr', 'diagnostics', 'default_config', 'sensor', 'input_text', 'device_tracker', 'renault', 'speedtestdotnet', 'map', 'onvif', 'group', 'huawei_solar', 'image', 'counter', 'input_number', 'ocpp', 'blitzortung', 'binary_sensor', 'dwd_weather', 'met', 'tts', 'frigate', 'input_datetime', 'octoprint', 'search', 'zone', 'config', 'script', 'localtuya', 'timer', 'dlna_dms', 'input_boolean', 'image_processing', 'sun', 'utility_meter', 'input_button', 'switch', 'repairs', 'mqtt', 'influxdb', 'application_credentials', 'lovelace', 'homeassistant_alerts', 'automation', 'tplink', 'logbook'}
2022-09-06 19:07:08.188 INFO (MainThread) [homeassistant.setup] Setting up solcast_solar
2022-09-06 19:07:08.188 INFO (MainThread) [homeassistant.setup] Setup of domain solcast_solar took 0.0 seconds
2022-09-06 19:07:09.202 DEBUG (MainThread) [custom_components.solcast_solar.coordinator] Solcast - Registering API auto fetching hourly between sun up and sun set
2022-09-06 19:07:09.202 DEBUG (MainThread) [custom_components.solcast_solar.coordinator] Solcast - API will only connect between the hours 6 and 20 and at midnight
2022-09-06 19:07:09.203 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast polling for data
2022-09-06 19:07:09.203 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast including past actual data? - True
2022-09-06 19:07:09.205 DEBUG (MainThread) [custom_components.solcast_solar.coordinator] Finished fetching solcast_solar data in 0.003 seconds (success: True)
2022-09-06 19:07:09.216 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.solcast_solar
2022-09-06 19:10:10.009 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast polling for data
2022-09-06 19:10:10.009 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast including past actual data? - True`

and using solcast_update_all_forecasts

`2022-09-06 19:30:23.226 DEBUG (MainThread) [custom_components.solcast_solar.coordinator] Solcast - Event called to force an update of data from Solcast API
2022-09-06 19:30:23.226 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast polling for data
2022-09-06 19:30:23.227 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast including past actual data? - False
File "/config/custom_components/solcast_solar/coordinator.py", line 145, in service_event_update`

Can I provide some more data? Anything else that I can try?

**Comment by oziee** [comment]
not sure how but GitHub did have old code that sneaked back in here, but not on my dev box

update to v3.0.20 and should be all good

**Comment by killee** [comment]
Thanks for the update.
But after updating, deinstalling, deleting, restart, clearing und reinstall the issu is still present:

`2022-09-07 15:02:47.976 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration solcast_solar which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-09-07 15:02:56.393 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 2: {'device_automation', 'hacs', 'onboarding', 'luftdaten', 'dwd_weather', 'input_button', 'media_source', 'image_processing', 'mobile_app', 'utility_meter', 'energy', 'octoprint', 'switch', 'person', 'met', 'automation', 'ocpp', 'timer', 'logbook', 'my', 'tag', 'input_datetime', 'blueprint', 'tts', 'speedtestdotnet', 'persistent_notification', 'frigate', 'shelly', 'system_health', 'co2signal', 'sensor', 'scheduler', 'alexa_media', 'application_credentials', 'trace', 'repairs', 'device_tracker', 'number', 'binary_sensor', 'maxcube', 'onvif', 'ffmpeg', 'homeassistant_alerts', 'localtuya', 'easycontrols', 'scene', 'dlna_dmr', 'analytics', 'input_text', 'mqtt', 'history', 'tplink', 'counter', 'forecast_solar', 'auth', 'camera', 'default_config', 'sun', 'group', 'input_number', 'zone', 'dlna_dms', 'lovelace', 'huawei_solar', 'esphome', 'config', 'solcast_solar', 'influxdb', 'image', 'map', 'diagnostics', 'script', 'renault', 'template', 'input_boolean', 'input_select', 'blitzortung', 'search'}
2022-09-07 15:02:57.748 INFO (MainThread) [homeassistant.setup] Setting up solcast_solar
2022-09-07 15:02:57.748 INFO (MainThread) [homeassistant.setup] Setup of domain solcast_solar took 0.0 seconds
2022-09-07 15:02:59.262 DEBUG (MainThread) [custom_components.solcast_solar.coordinator] Solcast - Registering API auto fetching hourly between sun up and sun set
2022-09-07 15:02:59.262 DEBUG (MainThread) [custom_components.solcast_solar.coordinator] Solcast - API will only connect between the hours 6 and 20 and at midnight
2022-09-07 15:02:59.263 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast polling for data
2022-09-07 15:02:59.263 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast including past actual data? - True
2022-09-07 15:02:59.263 DEBUG (MainThread) [custom_components.solcast_solar.coordinator] Finished fetching solcast_solar data in 0.001 seconds (success: True)
2022-09-07 15:02:59.280 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.solcast_solar
2022-09-07 15:06:53.281 DEBUG (MainThread) [custom_components.solcast_solar.coordinator] Solcast - Event called to force an update of data from Solcast API
2022-09-07 15:06:53.281 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast polling for data
2022-09-07 15:06:53.281 DEBUG (MainThread) [custom_components.solcast_solar.solcastapi] Solcast including past actual data? - False`

**Comment by oziee** [comment]
There is no error in the info you just posted? What error are you seeing?

**Comment by killee** [comment]
The same error since starting. No update of the data like the first post.

**Comment by Spookster** [comment]
With 3.0.20 I too woke to zero values.
Ive regenerated the API key and the values immediately returned

**Comment by Spookster** [comment]
I note its reset to 0 again, last polled 22 years.
What exact steps do you need us to take to give good debug information?

**Comment by oziee** [comment]
I am going to add more debug code to v21 so tracing might give more insight

**Comment by sexysaxon** [comment]
Hi Oziee, installed today but no forecast data  available. When checked logs get these error messages when restarting HA:

_Solcast http_data error: Traceback (most recent call last): File "/config/custom_components/solcast_solar/solcastapi.py", line 93, in sites_data raise Exception(f"HTTP error: Solcast Error gathering rooftop sites data.") Exception: HTTP error: Solcast Error gathering rooftop sites data.
13:48:23 – (ERROR) Solcast PV Forecast (custom integration)
Solcast Error 404 - Gathering rooftop sites data.
13:48:23 – (ERROR) Solcast PV Forecast (custom integration)_

I'm running Core 2022.9.2 on OS 8.4 or Raspberry Pi4

**Comment by oziee** [comment]
solcast has changed their API.. the roof_top site way is now marked as legacy.. 

I am currently reading through all the new api docs and will work on updating this to the new way

the old was is legacy and does still work. I am not seeing any of the errors you guys are seeing so its marking it hard to work out just what is going on.. It could have something to do with Solcast changing the api, even though it is still working ok, maybe at times its not, or while they were in the middle of changing their api system things started failing.. who knows

the new integration will be marked as version 4 just to make things easy for all knowing who is using what code etc for issues etc

give me a few days to work through it all.. I also only get 50 api calls so while coding and debugging I sometimes use all my 50 up quickly without thinking and then I have to wait for it to rest to continue testing ;)

**Comment by oziee** [comment]
v3.0.21 is up.. to help with any more debug log info to track and see if we can work this out

**Comment by killee** [comment]
Thanks @oziee
I've attached the log. If I can help to debug give me a hint.

[log.txt.txt](https://github.com/oziee/ha-solcast-solar/files/9606687/log.txt.txt)

**Comment by pgwg-uk** [comment]
I'm seeing the same issue here, however I'm also getting 403 errors when querying the API via postman or node-red as well.
FYI, if you're looking to query against the API for testing purposes, you can query the following lat / long without using up the free daily allowance.

Latitude | Longitude | Location
-33.856784 | 151.215297 | Sydney Opera House
36.099763 | -112.112485 | Grand Canyon
51.178882 | -1.826215 | Stonehenge
41.89021 | 12.492231 | The Colosseum
29.977296 | 31.132496 | Giza Pyramid Complex
27.175145 | 78.042142 | Taj Mahal

**Comment by sexysaxon** [comment]
Hi Ossie, installed v3.0.21 but still same error as previously

Solcast http_data error: Traceback (most recent call last): File "/config/custom_components/solcast_solar/solcastapi.py", line 95, in sites_data raise Exception(f"HTTP error: Solcast Error gathering rooftop sites data.") Exception: HTTP error: Solcast Error gathering rooftop sites data.
15:51:19 – (ERROR) Solcast PV Forecast (custom integration)
Solcast Error 404 - Gathering rooftop sites data.
15:51:19 – (ERROR) Solcast PV Forecast (custom integration)

**Comment by oziee** [comment]
@sexysaxon in the log somewhere it will log out the url it's trying to get to

Find that and put it in a browser and see what happens

A 404 error is page not found so either the url is wrong or solcast is having problems

Someone the other day was getting 403 errors so again that is a solcast problem

They seem to be changing some things and not sure why some ppl have zero problems and other all problems.. so hopefully the url debug line can help

**Comment by sexysaxon** [comment]
Hi ozieee, can't find any reference to url in logs. I've restarted gain so can show you full log below, hope this helps.

Full logs
2022-09-23 09:16:13.149 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration solcast_solar which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-09-23 09:16:13.163 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hildebrandglow_dcc which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-09-23 09:16:13.167 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-09-23 09:16:13.169 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration octopus_energy which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-09-23 09:16:35.434 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Canon MX530 series' for ipp integration not ready yet: Invalid response from API: Timeout occurred while connecting to IPP server.; Retrying in background
2022-09-23 09:16:36.326 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform octopus_energy is taking over 10 seconds.
2022-09-23 09:16:36.931 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Shelly EM PV and Grid Monitor' for shelly integration not ready yet: Timeout during device setup; Retrying in background
2022-09-23 09:16:36.969 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Shelly EM Battery Monitor' for shelly integration not ready yet: Timeout during device setup; Retrying in background
2022-09-23 09:16:37.147 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] Solcast Error 404 - Gathering rooftop sites data.
2022-09-23 09:16:37.161 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] Solcast http_data error: Traceback (most recent call last):
File "/config/custom_components/solcast_solar/solcastapi.py", line 95, in sites_data
raise Exception(f"HTTP error: Solcast Error gathering rooftop sites data.")
Exception: HTTP error: Solcast Error gathering rooftop sites data.
2022-09-23 09:16:37.412 WARNING (MainThread) [homeassistant.components.media_player] Setup of media_player platform dlna_dmr is taking over 10 seconds.
2022-09-23 09:16:37.413 WARNING (MainThread) [homeassistant.components.media_player] Setup of media_player platform dlna_dmr is taking over 10 seconds.
2022-09-23 09:16:46.611 ERROR (MainThread) [homeassistant.helpers.event] Error while processing template: Template("{%set solarpv = states('sensor.power_monitor_channel_1_energy') | float%} {%set feedinrate = 0.62%} {{ solarpv * feedinrate }}")
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 1677, in forgiving_float_filter
return float(value)
ValueError: could not convert string to float: 'unavailable'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 419, in async_render
render_result = _render_with_context(self.template, compiled, **kwargs)
File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 1904, in _render_with_context
return template.render(**kwargs)
File "/usr/local/lib/python3.10/site-packages/jinja2/environment.py", line 1301, in render
self.environment.handle_exception()
File "/usr/local/lib/python3.10/site-packages/jinja2/environment.py", line 936, in handle_exception
raise rewrite_traceback_stack(source=source)
File "<template>", line 1, in top-level template code
File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 1680, in forgiving_float_filter
raise_no_default("float", value)
File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 1393, in raise_no_default
raise ValueError(
ValueError: Template error: float got invalid input 'unavailable' when rendering template '{%set solarpv = states('sensor.power_monitor_channel_1_energy') | float%} {%set feedinrate = 0.62%} {{ solarpv * feedinrate }}' but no default was specified
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 535, in async_render_to_info
render_info._result = self.async_render(variables, strict=strict, **kwargs)
File "/usr/src/homeassistant/homeassistant/helpers/template.py", line 421, in async_render
raise TemplateError(err) from err
homeassistant.exceptions.TemplateError: ValueError: Template error: float got invalid input 'unavailable' when rendering template '{%set solarpv = states('sensor.power_monitor_channel_1_energy') | float%} {%set feedinrate = 0.62%} {{ solarpv * feedinrate }}' but no default was specified
2022-09-23 09:16:46.627 ERROR (MainThread) [homeassistant.helpers.template_entity] TemplateError('ValueError: Template error: float got invalid input 'unavailable' when rendering template '{%set solarpv = states('sensor.power_monitor_channel_1_energy') | float%} {%set feedinrate = 0.62%} {{ solarpv * feedinrate }}' but no default was specified') while processing template 'Template("{%set solarpv = states('sensor.power_monitor_channel_1_energy') | float%} {%set feedinrate = 0.62%} {{ solarpv * feedinrate }}")' for attribute '_attr_native_value' in entity 'sensor.fit_payment'

**Comment by oziee** [comment]
> I'm seeing the same issue here, however I'm also getting 403 errors when querying the API via postman or node-red as well. I'm seeing this when querying both the old and new endpoints.
> 
> FYI, if you're looking to query against the API for testing purposes, you can query the following lat / long without using up the free daily allowance.
> 
> Latitude | Longitude | Location -33.856784 | 151.215297 | Sydney Opera House 36.099763 | -112.112485 | Grand Canyon 51.178882 | -1.826215 | Stonehenge 41.89021 | 12.492231 | The Colosseum 29.977296 | 31.132496 | Giza Pyramid Complex 27.175145 | 78.042142 | Taj Mahal

nar I just download a copy of the json data and spin up a python web server and point ha integration to my laptop.. I just sometimes forget to set the web address back to solcast from time to time:)

**Comment by oziee** [comment]
added more debug code

**Comment by killee** [comment]
> 

Thanks, the log is attached

[log_solcast2.txt](https://github.com/oziee/ha-solcast-solar/files/9643840/log_solcast2.txt)

**Comment by oziee** [comment]
@killee looking at the log it seems that Solcast is not listing the sites you have under that API key

in the log file on line 
`(MainThread) [custom_components.solcast_solar.solcastapi] SOLCAST: trying to connect to - https://api.solcast.com.au/rooftop_sites?format=json&api_key=yLWPuBpq3w...........................`

if you copy and paste that url into a browser window what is the data solcast returns?? it should be a json of the sites you have created in the toolkit

but I see this in the debug code
```
SOLCAST: sites_data code got data {'sites': [], 'page_count': 0, 'current_page': 1, 'total_records': 0}
```

it does not list any sites its blank `[]`
when I load my line from the log file with my API key I see this
<img width="434" alt="Screen Shot 2022-09-26 at 6 32 27 pm" src="https://user-images.githubusercontent.com/1471841/192230623-9d7fdd70-3d89-4c29-ae0e-64fd489131c7.png">

being that solcast it not returning any data, you need to check that your API key is valid for the toolkit with sites created with in it

**Comment by killee** [comment]
Thanks @oziee got it.
I've did the setup of solcast wrong. I've created a test api at: https://toolkit.solcast.com.au/world-api
But I have to create a rooftop under: https://toolkit.solcast.com.au/rooftop-sites/create

So it is a user fail. But perhaps some of the others have similar issues!

**Comment by killee** [comment]
> @killee make sure to delete the demo items otherwise they will show up in the data

Tanks, get it now to work like a charm!
Thanks for your support and for the great work!

Sorry for the extra work

**Comment by sexysaxon** [comment]
Hi Ozziee, this still hasn't fixed my issue. I believe my Solcast profile is configured correctly as I have been using the same one using the same API key on other applications.
Here are what I believe are the relevant logs:
2022-10-03 15:31:40.779 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration solcast_solar which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-10-03 15:32:06.097 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] Solcast Error 404 - Gathering rooftop sites data.
2022-10-03 15:32:06.106 ERROR (MainThread) [custom_components.solcast_solar.solcastapi] Solcast http_data error: Traceback (most recent call last):
File "/config/custom_components/solcast_solar/solcastapi.py", line 96, in sites_data
raise Exception(f"HTTP error: Solcast Error gathering rooftop sites data.")
Exception: HTTP error: Solcast Error gathering rooftop sites data.
2022-10-03 16:29:26.750 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration solcast_solar which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
